cpp/src/io/parquet/page_data.cu

```diff
@@ -1657,7 +1658,9 @@ extern "C" __global__ void __launch_bounds__(NTHREADS)
       // - updates offsets (for nested columns)
       // - produces non-NULL value indices in s->nz_idx for subsequent decoding
       gpuDecodeLevels(s, target_pos, t);
-    } else if (t < out_thread0) {
+    }
+    __syncthreads();
```

Comment:
`s->nz_idx` is getting updated in `gpuDecodeLevels`, which is again read and updated in 1675, so had to split.

Reply:
In that case it might not be required if decode step is working on iteration N-1. It would just a False positive. Would wait for @OlivierNV

Reply:
Yeah, that syncthreads() just serializes gpuDecodeLevels() and the rest of the decoding, which sort of defeats the point of using different warps to do different things.
Also, all the _syncwarps() in this PR are basically WARs for the cuda tool that doesn't treat shfl_sync or ballot_sync as synchronization primitives (but that's fundamentally a tool bug imo).